site/source/tutorials/005_form_widgets/index.md

```diff
@@ -1,26 +1,159 @@
+
+The last change that needs to be made in the `WorkerForm` is to update the `_onSubmit` method to delegate to the `onSubmit` property when it is called. Replace that method with this:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerForm.ts' lines:26-31 %}
+
+## Integrating the WorkerForm into the App
+
+Now that the `WorkerForm` widget is done, it's time to update the `App` class to take advantage of it. First, we need to address how to store the form's data as the user fills it out. Recall that the `WorkerForm` will accept an `onChange` property that will allow the `App` class to be informed when a field value changes. However, the `App` class doesn't currently have a place to store those changes. We could add private fields in the class to store those values, but that would be difficult to maintain over time as the application gets more complicated and more data needs to be stored. Instead, we are going to take advantage of another mixin that comes with Dojo 2 that is designed to handle state for us - the `StatefulMixin`.
+
+The `StatefulMixin` allows a class to store data in an object that can be updated by calling the `setState` method. Calling this method will not only update the object's state, but it will also raise an event that will trigger the application to re-render the page to reflect the new state.
+
+Import the `StatefulMixin` by adding this line to the top of `App.ts`:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' line:4 %}
+
+And then add `AppBase` and update the class declaration:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' lines:10-12 %}
+
+Now, update the `render` method to populate the `WorkerForm`'s properties:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' lines:73-110 %}
+
+Most of the properties are being populated by the `formData` object that is being retrieved from the `App`'s state with the only exceptions being the event handlers.
+
+The `onChange` handler is calling the `App`'s `setState` method, passing in all of the existing state (via the `...this.state.formData` statement), and appending the new state data coming in from the event (via `[field]: value`). After the event is processed, Dojo 2 will automatically detect that the state has changed and call the `invalidate` method, causing the page to be refreshed with the new data.
+
+The `onBlur` handler is almost the same as the `onChange` handler, except that it is making a call to the `_validateWorkerData` method and the result to update the `state`'s `formData`. Add that method to the `App` class as follows:
```

Comment:
"The `onBlur` handler is almost the same as the `onChange` handler, except that it is making a call to the `_validateWorkerData` method and the result to update the `state`'s `formData`. Add that method to the `App` class as follows:" -> "The `onBlur` handler is nearly identical to the `onChange` handler, except that it calls the `_validateWorkerData` method and uses the returned result to update the `state`'s `formData`. Add that method to the `App` class:"  ... also, replace the last that with the name of the actual method